opafiber/fiber.go

```diff
@@ -9,66 +9,62 @@ import (
 	"io"
 )
 
+type InputCreationFunc func(c *fiber.Ctx) (map[string]interface{}, error)
+
 type Config struct {
 	RegoPolicy            io.Reader
 	RegoQuery             string
 	IncludeHeaders        []string
 	IncludeQueryString    bool
 	DeniedStatusCode      int
 	DeniedResponseMessage string
+	InputCreationMethod   InputCreationFunc
 }
 
 func New(cfg Config) fiber.Handler {
-	if cfg.RegoQuery == "" {
-		panic(fmt.Sprint("rego query can not be empty"))
-	}
-
-	if cfg.DeniedStatusCode == 0 {
-		cfg.DeniedStatusCode = fiber.StatusBadRequest
+	err := cfg.fillAndValidate()
+	if err != nil {
+		panic(err)
 	}
-
 	readedBytes, err := io.ReadAll(cfg.RegoPolicy)
 	if err != nil {
 		panic(fmt.Sprint("could not read rego policy %w", err))
 	}
-
 	query, err := rego.New(
 		rego.Query(cfg.RegoQuery),
 		rego.Module("policy.rego", utils.UnsafeString(readedBytes)),
 	).PrepareForEval(context.Background())
-
 	if err != nil {
 		panic(fmt.Sprint("rego policy error: %w", err))
 	}
-
 	return func(c *fiber.Ctx) error {
-		input := map[string]interface{}{
-			"method": c.Method(),
-			"path":   c.Path(),
+		input := make(map[string]interface{})
+		input, err = cfg.InputCreationMethod(c)
+		if err != nil {
+			c.Response().SetStatusCode(fiber.StatusInternalServerError)
+			c.Response().SetBodyString(fmt.Sprintf("Error creating input: %s", err))
+			return err
 		}
-
 		if cfg.IncludeQueryString {
 			queryStringData := make(map[string][]string)
 			c.Request().URI().QueryArgs().VisitAll(func(key, value []byte) {
 				queryStringData[utils.UnsafeString(key)] = append(queryStringData[utils.UnsafeString(key)], utils.UnsafeString(value))
 			})
 			input["query"] = queryStringData
 		}
-
-		if len(cfg.IncludeHeaders) > 0 {
-			headers := make(map[string]string)
-			for _, header := range cfg.IncludeHeaders {
-				headers[header] = c.Get(header)
-			}
-			input["headers"] = headers
+		headers := make(map[string]string)
```

Comment:
It creates a map even if the user does not include headers? I think the "if" condition should remain, WDYT @ReneWerner87 @efectn ?